opencensus/metrics/transport.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import threading
+
+from opencensus.common import utils
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_INTERVAL = 60
+GRACE_PERIOD = 5
+
+
+class TransportError(Exception):
+    pass
+
+
+class PeriodicTask(threading.Thread):
```

Comment:
This could be useful for other exporters (e.g. trace exporters), please consider putting it in `opencensus.common`.

Reply:
For trace exporters we want a queue that periodically exports batches of spans. It's possible that we'd reuse this, but I think we ought to move it when we do instead of now.

Reply:
Make sense.